pkg/util/targz/targz.go

```diff
@@ -16,29 +16,28 @@ package targz
 import (
 	"archive/tar"
 	"bytes"
-	"compress/gzip"
 	"io"
 	"io/fs"
 	"path/filepath"
 
 	"github.com/cockroachdb/errors"
+	"github.com/klauspost/compress/zstd"
 	"github.com/spf13/afero"
 )
 
 // AsFS exposes the contents of the given reader (which is a .tar.gz file)
```

Comment:
As above, what is here is no longer a .tar.gz file. Please take a pass for all the instances of `.tar.gz` in this file and replace them.

Reply:
I was under the impression that zstd encodes (or can encode) in `gz` format. Not sure if this is what this library is doing though.

Reply:
Good point. I didn't set `--format=gzip`; by default, it uses its own header with a magic that gzip doesn't recognize. Let me see if the compatible format would change anything. If that does work, I don't have to find a new name, so it's a win-win :)

Reply:
I don't particularly care whether we use `gzip` or not, just want to make sure that the comments and documentation all agree with the actual functionality of the thing.

Reply:
`klauspost/compress/zstd` doesn't support `gzip` headers afaik, so I renamed everything to `assetbundle`, as suggested. 

@rickystewart PTAL. I also think that we should strengthen the checks in `bincheck` by asserting conservatively on the total `init` time and heap allocations. This way, we can fail a build at the time of a suspicious change vs. having to discover this much later.

Reply:
I am not sure how we would implement that besides e.g. checking how much time `cockroach version` takes to run. If you have a brighter idea, I'd like to hear it.

`bincheck` is not run during CI, but only during the release process. (#78999 was closed by the bot but to my knowledge was never actually done.) So keep that in mind.

Reply:
I've made the change to `bincheck`; the two checks use fairly conservative upper-bounds, so my hope is that should be enough _not_ to flake and to catch unexpected changes to the startup time and memory footprint.

https://github.com/cockroachdb/cockroach/pull/127292